managedtenants/bundles/index_builder.py

```diff
@@ -26,7 +26,11 @@ def __init__(
         )
 
     def build_and_push(
-        self, bundles, hash_string=get_short_hash(), skip_validation=False
+        self,
+        bundles,
+        hash_string=get_short_hash(),
+        skip_validation=False,
+        test_run=False,
```

Comment:
Kinda not sold on the need of the extra arg of `test_run` to be honest where the `skip_validation` seems to be capable of accomplishing the same purpose :confused:
They both seem to duplicate each other, thereby unnecessarily complicating the API.

If semantics were your concern, then I think it would be better to just rename `skip_validation` to `dry_run` and use that instead but again, that also seems a bit far-fetched.

Or Am I missing something here and we do have a place where test_run might have a dedicated purpose like, `skip_validation=True` and `test_run=False` or vice versa?

Would love to know your thoughts on this and feel free to correct me wherever the need be :slightly_smiling_face:

Thanks!

Reply:
Thanks for pointing this out. I just noticed the `_validate_sql_catalog` part and `skip_validation` being dedicately used for that.
I guess the term "skip_validation" caused me to presume that it is more of a dry-run-oriented thing rather than specifically being dedicated to catalog validation.

Now I see the point of `test_run` although I would suggest a different for it. For example, instead of `test_run=False` maybe we can use `use_digest=True`.

```
              [
                    bundle.image.url_digest
                    if use_digest
                    else bundle.image.url_tag
                    for bundle in bundles
                ]
```
Because right now, the only use-case of test_run param is to signify to use url_tag or url_digest, instead of any diffenent test_run-oriented functional logic. (for example, skipping builds or modifying build procedure entirely).

In the future, if we wanna have a test_run-oriented logic in this method, then, we can either use an env var or a `dry_run` to signify that.

Let me know what do you think :)

Reply:
Update on this: @rpriyanshu9 pointed out the actual reason behind the skip_validation param i.e. it is to skip the catalog-level validation of bundles.

Reply:
Thanks for pointing this out. I just noticed the `_validate_sql_catalog` part and `skip_validation` being dedicately used for that.
I guess the term "skip_validation" caused me to presume that it is more of a dry-run-oriented thing rather than specifically being dedicated to catalog validation.

Now I see the point of `test_run` although I would suggest a different for it. For example, instead of `test_run=False` maybe we can use `use_digest=True`.

```
              [
                    bundle.image.url_digest
                    if use_digest
                    else bundle.image.url_tag
                    for bundle in bundles
                ]
```
Because right now, the only use-case of test_run param is to signify to use url_tag or url_digest, instead of any diffenent test_run-oriented functional logic. (for example, skipping builds or modifying build procedure entirely).

In the future, if we wanna have a test_run-oriented logic in this method, then, we can either use an env var or a `dry_run` to signify that.

Let me know what do you think :slightly_smiling_face:

Reply:
My problem with test paths in the code is that you don't test what you run. instead, we should create mockups and execute the actual code. for instance, if a new version of the sretoolbox changes/breaks the `url_digest()`, we won't know until it's too late.